src/core/pointcloud/expression/qgspointcloudexpression.h

```diff
@@ -205,6 +205,13 @@ class CORE_EXPORT QgsPointCloudExpression
      */
     static bool checkExpression( const QgsExpression &expression, const QgsPointCloudBlock *block, QString &errorMessage );
 
+    /**
+     * Returns an expression string converted to PDAL expression format.
+     *
+     * \since QGIS 3.32
+     */
+    QString asPdalExpression() const;
```

Comment:
I think this needs some `bool& ok SIP_OUT, QString& error SIP_OUT` arguments so that callers can get feedback if an expression can't be converted to PDAL format